### PR TITLE
GitHub actions update

### PR DESCRIPTION
**Pull request by mzappitello**
Correct the Deploy Script dependencies and add tests on a potential action to only test what we need to test.

### CODE REVIEW

File: .github/workflows/future_continuous_integration.yaml
Comment (mzappitello) [inline .github/workflows/future_continuous_integration.yaml:1]: this is an outline of how we can figure out what tests we want to run on each pull request based on whats changed. it uses bash commands to parse out what directories have changed and we can us the bools it generates to determine what needs to be run.

File: .github/workflows/future_continuous_integration.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      if: env.TEST_AWS == 'true'
+      run: echo "Running AWS tests..."
+
+    - name: Test Performance Manager
+      if: env.TEST_PERFORMANCE_MANAGER == 'true'
+      run: echo "Running Performance Manager tests..."
+
+    - name: Test Postgres 
+      if: env.TEST_POSTGRES == 'true'
+      run: echo "Running Postgres tests..."
```
Comment (mzappitello) [inline .github/workflows/future_continuous_integration.yaml:91]: the idea is that this section will eventually have `poetry run pytest tests/aws` or whatever. we'll also standup the postgres server only for the tests that require them.

File: .github/workflows/ingestion-deploy-dev.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ name: Deploy Ingestion to Dev
 
 on:
   workflow_run:
-    workflows: ["Ingestion CI"]
+    workflows: ["Continuous Integration"]
```
Comment (mzappitello) [inline .github/workflows/ingestion-deploy-dev.yaml:5]: this needs to be tied to the ci passing and the main branch.

File: .github/workflows/future_continuous_integration.yaml
Comment (mzappitello) [reply .github/workflows/future_continuous_integration.yaml:1]: output of this is here https://github.com/mbta/lamp/actions/runs/4441953860

File: .github/workflows/future_continuous_integration.yaml (hunk 1 above)
Comment (rymarczy) [reply .github/workflows/future_continuous_integration.yaml:91]: I'm not totally sure what all is happening in "Check Changed Subdirectories".

It does look there are some hard coded file names and command parameters that we need to be very aware of, which changes to folder structure in the project. Might be worth adding that information as comments and also maybe some instructions on how to handle the additions of new library folders.

**Comment by rymarczy** [comment]
Main changes to deployment look good. Not too worried about "future_continuous_integration" since it won't be running as our main CI right now.

**Comment by mzappitello** [comment]
i moved the new CI stuff out into its own PR

https://github.com/mbta/lamp/pull/101

i also took the notes here when doing so.

File: .github/workflows/deploy-dev.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+name: Deploy Ingestion to Dev
```
Comment (rymarczy) [inline .github/workflows/deploy-dev.yaml:1]: Should this be a more general name?

**Comment by rymarczy** [comment]
A couple of comments. Fixing the Dockerfile would be the only blocked on this.

**Comment by rymarczy** [comment]
LGTM

**Pull request merged**